src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -3008,20 +3008,15 @@ bool sockinfo_tcp::is_readable(uint64_t *p_poll_sn, fd_array_t* p_fd_array)
 	int ret;
 
 	if (is_server()) {
-		bool state;
-		//tcp_si_logwarn("select on accept()");
-		//m_conn_cond.lock();
-		state = m_ready_conn_cnt == 0 ? false : true; 
-		if (state) {
+		if (m_ready_conn_cnt) {
 			si_tcp_logdbg("accept ready");
-			goto noblock_nolock;
+			return true;
 		}
 
-		if (m_sock_state == TCP_SOCK_ACCEPT_SHUT) goto noblock_nolock;
-
-		return false;
+		return m_sock_state == TCP_SOCK_ACCEPT_SHUT ? true : false;
```

Comment:
could you consider wrapping with (...) to improve visibility (it is not mandatory)